dipy/reconst/dki.py

```diff
@@ -28,6 +28,459 @@
 from .base import ReconstModel
 
 
+def _positive_evals(L1, L2, L3, er=None):
+    """ Helper function that indentifies which voxels in a array have all
+    eigenvalues larger than zero
+
+    Parameters
+    ----------
+    L1 : ndarray (n,)
+        First independent variable of the integral.
+    L2 : ndarray (n,)
+        Second independent variable of the integral.
+    L3 : ndarray (n,)
+        Third independent variable of the integral.
+    er : float, optional
+        A eigenvalues is classified as larger than zero if it is larger than er
+
+    Returns
+    -------
+    ind : boolean (n,)
+        Array that marks the voxels that have all eigenvalues are larger than
+        zero.
+    """
+    if er is None:
+        er = np.finfo(L1[0]).eps * 1e3
```

Comment:
Once you do that. I believe that all the other functions below could also be written to operate on arbitrary-shape ndarrays. In particular, check out `from dipy.core.ndindex import ndindex` to iterate over arbitray-shape arrays